overlord/devicestate/devicemgr.go

```diff
@@ -371,6 +374,48 @@ func (m *DeviceManager) ensureBootOk() error {
 	return nil
 }
 
+func markSeededInConfig(st *state.State) error {
+	var seedDone bool
+	tr := config.NewTransaction(st)
+	if err := tr.Get("core", "seed.loaded", &seedDone); err != nil && !config.IsNoOption(err) {
+		return err
+	}
+	if !seedDone {
+		if err := tr.Set("core", "seed.loaded", true); err != nil {
+			return err
+		}
+		tr.Commit()
+	}
+	return nil
+}
+
+func (m *DeviceManager) ensureSeedInConfig() error {
```

Comment:
Do we need new method?  We already have a place changing "seeded" itself in state.. is that not an appropriate place for changing the configuration?

Reply:
we also need to cover preexisting systems though,  so we need something in one of the ensure handlers for that

Reply:
Thanks @niemeyer - I pushed https://github.com/snapcore/snapd/pull/5157 that adds a better explanation and some more cleanups/tweaks for this PR (some more unit tests etc).